src/frontend/js/components/DashboardItem/DashboardItemCourseEnrolling.tsx

```diff
@@ -0,0 +1,288 @@
+    id: 'components.DashboardItemCourseEnrollingRun.enrollmentNotYetOpened',
+  },
+  noCourseRunAvailable: {
+    defaultMessage: 'No session available for this course.',
+    description: 'Text displayed when no course run are opened for the course',
+    id: 'components.DashboardItemCourseEnrollingRun.noCourseRunAvailable',
+  },
+};
+
+interface Props {
+  course: AbstractCourse;
+  activeEnrollment?: Enrollment;
+  order?: Order;
+  writable?: boolean;
+  icon?: boolean;
+  notEnrolledUrl?: string;
+}
+
+export const DashboardItemCourseEnrolling = ({
+  course,
+  activeEnrollment,
+  writable,
+  order,
+  icon = false,
+  notEnrolledUrl = '#',
+}: Props) => {
+  return (
+    <div data-testid={'dashboard-item__course-enrolling__' + course.code}>
+      {!writable && (
+        <div className="dashboard-item__course-enrolling__infos">
+          {!!activeEnrollment && <Enrolled icon={icon} enrollment={activeEnrollment} />}
+          {!activeEnrollment && <NotEnrolled icon={icon} notEnrolledUrl={notEnrolledUrl} />}
+        </div>
+      )}
+      {writable && (
+        <DashboardItemCourseEnrollingRuns
+          course={course}
+          enrollments={CoursesHelper.findCourseEnrollmentsInOrder(course, order!)}
+          order={order}
+        />
+      )}
+    </div>
+  );
+};
+
+const DashboardItemCourseEnrollingRuns = ({
+  course,
+  enrollments,
+  order,
+}: {
+  course: AbstractCourse;
+  enrollments: Enrollment[];
+  order?: Order;
+}) => {
+  const { enroll, isLoading, error } = useEnroll(enrollments, order);
+
+  // Hide runs with finished enrollment.
+  const datas = useMemo(
+    () =>
+      course.course_runs
+        .map((courseRun) => ({
+          courseRun,
+          selected: !!enrollments.find(
+            (enrollment) =>
+              courseRun.resource_link === enrollment.course_run.resource_link &&
+              enrollment.is_active,
+          ),
+        }))
+        .filter(
+          (data) => data.selected || data.courseRun.state.priority <= Priority.FUTURE_NOT_YET_OPEN,
+        ),
+    [course, enrollments],
+  );
+
+  // Filter by priority
+  return (
+    <div className="dashboard-item__course-enrolling__runs">
+      {error && <Banner message={error} type={BannerType.ERROR} />}
+      {datas.length === 0 && (
+        <div className="dashboard-item__course-enrolling__no-runs">
+          <Icon name="icon-warning" size="small" />
+          <FormattedMessage {...messages.noCourseRunAvailable} />
+        </div>
+      )}
+      {datas.map((data) => (
+        <DashboardItemCourseEnrollingRun
+          key={data.courseRun.id}
+          courseRun={data.courseRun}
+          selected={data.selected}
+          enroll={() => enroll(data.courseRun)}
+        />
+      ))}
+      {isLoading && (
+        <div
+          className="dashboard-item__course-enrolling__loading"
+          data-testid="dashboard-item__course-enrolling__loading"
+        >
+          <Spinner size="large" />
```

Comment:
For a11y, this spinner should have an `aria-labelledby` attribute.
e.g:

```suggestion
          <Spinner aria-labelledby="enrolling-loading" size="large">
            <span id="enrolling-loading">
              <FormattedMessage {...messages.loading} />
            </span>
          <Spinner/>
```